internal/controller/server_controller.go

```diff
@@ -512,7 +513,7 @@ func (r *ServerReconciler) ensureServerPowerState(ctx context.Context, log logr.
 		return nil
 	}
 
-	bmcClient, err := GetBMCClientFromBMCName(ctx, r.Client, server.Spec.BMCRef.Name, r.Insecure)
+	bmcClient, err := GetBMCClientForServer(ctx, r.Client, server, r.Insecure)
```

Comment:
Just curious, if it's intentional, this should be "FromServer" or "ForServer", both are semantically correct though.

Reply:
I somehow found `For` a better naming here 🙈